elasticsearch/client/init.sls

```diff
@@ -7,16 +7,32 @@ include:
 {%- for index_name, index in client.get('index', {}).iteritems() %}
 elasticsearch_index_{{ index_name }}:
   {%- if index.get('enabled', False) %}
-  elasticsearch_index_template.present:
+  {% set operation = 'create' %}
+  {%- if index.definition is defined %}
+  {% set definition = index.definition %}
+  {%- else %}
+  {%- if index.template is defined %}
   {%- import_json index.template as definition %}
+  {%- endif %}
```

Comment:
if neither template nor definition are defined, it will fail badly. you need to add a test before ``elasticsearch_index_{{ index_name }}``.